src/snowflake/connector/s3_storage_client_async.py

```diff
@@ -0,0 +1,364 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import asyncio
+
+import aiohttp
+
+from .s3_storage_client import *
+from .storage_client_async import *
+
+
+# YICHUAN: Prioritize methods inherited SnowflakeStorageClientAsync because it will only override what it needs to
+# For the most part, this class overrides private methods with "_async" suffix that will be called by public methods
+# in SnowflakeStorageClientAsync
+class SnowflakeS3RestClientAsync(SnowflakeStorageClientAsync, SnowflakeS3RestClient):
+    def transfer_accelerate_config(
+        self, use_accelerate_endpoint: bool | None = None
+    ) -> bool:
+        # if self.endpoint has been set, e.g. by metadata, no more config is needed.
+        if self.endpoint is not None:
+            return self.endpoint.find("s3-accelerate.amazonaws.com") >= 0
+        if self.use_s3_regional_url:
+            self.endpoint = (
+                f"https://{self.s3location.bucket_name}."
+                f"s3.{self.region_name}.amazonaws.com"
+            )
+            return False
+        else:
+            if use_accelerate_endpoint is None:
+                use_accelerate_endpoint = self._loop_runner.run_coro(
+                    self._get_bucket_accelerate_config_async(
+                        self.s3location.bucket_name
+                    )
+                )
+
+            if use_accelerate_endpoint:
+                self.endpoint = (
+                    f"https://{self.s3location.bucket_name}.s3-accelerate.amazonaws.com"
+                )
+            else:
+                self.endpoint = (
+                    f"https://{self.s3location.bucket_name}.s3.amazonaws.com"
+                )
```

Comment:
Is this easier to read?
```suggestion
            self.endpoint = (
                f"https://{self.s3location.bucket_name}."
                + (
                    "s3-accelerate.amazonaws.com" if use_accelerate_endpoint
                    else "s3.amazonaws.com"
                )
            )
```
Maybe not...